src/base/QXmppTrustMessageElement.cpp

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (C) 2008-2021 The QXmpp developers
+ *
+ * Author:
+ *  Melvin Keskin
+ *
+ * Source:
+ *  https://github.com/qxmpp-project/qxmpp
+ *
+ * This file is a part of QXmpp library.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ */
+
+#include "QXmppTrustMessageElement.h"
+
+#include <QDomElement>
+
+#include "QXmppConstants_p.h"
+
+///
+/// \class QXmppTrustMessageElement
+///
+/// \brief The QXmppTrustMessageElement class represents a trust message element
+/// as defined by \xep{0434} Trust Messages (TM).
+///
+/// \since QXmpp 1.5
+///
+
+class QXmppTrustMessageElementPrivate : public QSharedData
+{
+public:
+    QString usage;
+    QString encryption;
+    QList<QXmppTrustMessageKeyOwner> keyOwners;
+};
+
+///
+/// Constructs a trust message element.
+///
+QXmppTrustMessageElement::QXmppTrustMessageElement() : d(new QXmppTrustMessageElementPrivate) {}
+
+///
+/// Constructs a copy of \a other.
+///
+/// \param other
+///
+QXmppTrustMessageElement::QXmppTrustMessageElement(const QXmppTrustMessageElement &other) : d(other.d) {}
+
+QXmppTrustMessageElement::~QXmppTrustMessageElement() = default;
+
+///
+/// Assigns \a other to this trust message element.
+///
+/// \param other
+///
+QXmppTrustMessageElement &QXmppTrustMessageElement::operator=(const QXmppTrustMessageElement &other)
+{
+    d = other.d;
+    return *this;
+}
+
+///
+/// Returns the namespace of the protocol that is intended to make use of the
+/// trust message.
+///
+/// \return the usage namespace
+///
+QString QXmppTrustMessageElement::usage() const
+{
+    return d->usage;
+}
+
+///
+/// Sets the namespace of the protocol that is intended to make use of the
+/// trust message.
+///
+/// \param usage usage namespace
+///
+void QXmppTrustMessageElement::setUsage(const QString &usage)
+{
+    d->usage = usage;
+}
+
+///
+/// Returns the namespace of the encryption protocol that uses the keys that are specified by
+/// the key identifiers of the trust message.
```

Comment:
One question: What's the difference between usage and encryption?

And can simplify this sentence a bit, it looks a bit complicated?